spec/dnsmasq_spec.rb

```diff
@@ -0,0 +1,24 @@
+RSpec.describe Dnsmasq do
+  let(:dnsmasq_instance) {
+    # Mocking the machine class
+    machine = Class.new do
```

Comment:
rather than an anonymous class you should use an rspec double -- see the docs at https://github.com/rspec/rspec-mocks , or an example in `spec/docker_spec.rb` in this project